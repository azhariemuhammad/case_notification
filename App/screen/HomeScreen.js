import React from 'react'
import {StyleSheet} from 'react-native'
import {Content, Container, Card, CardItem, Body, Text, Button} from "native-base";
import BoxHeader from "../component/BoxHeader";
import CardOption from "../component/CardOption";
import SentBoxScreen from "./SentBoxScreen";
import {baseService} from "../services";


class HomeScreen extends React.Component {
    static navigationOptions = {
        header: null
    };

    state={
        facilityName: '',
        facilityLevel: null
    }

    async componentDidMount() {
        await baseService().getUserDetail().then(res => {
            console.log(res.data)
            if (res.data.health_facility_name) {
                const {health_facility_name, facility_level} = res.data
                this.setState({facilityName: health_facility_name, facilityLevel: facility_level})
            }
        }).catch(e => {
            console.log('Somenthing went wrong', e)
        })
    }

    render() {
        return (
            <Container>
                <BoxHeader title={this.state.facilityName}/>
                <Content>
                    <CardOption title={"ACD"} body={"Active Case Detection"}/>
                    <CardOption title={"PCD"} body={"Passive Case Detection"}/>
                    {(this.state.facilityLevel === '1') ?
                        <Text style={styles.btnLink} onPress={
                            () => this.props.navigation.navigate('SentBoxScreen')
                        }>Lihat Laporan Terkirim</Text>
                        :
                        <Text style={styles.btnLink} onPress={
                            () => this.props.navigation.navigate('SentBoxScreen')
                        }>Lihat Laporan Diterima</Text>
                    }

                </Content>

            </Container>
        )
    }
}


const styles = StyleSheet.create({
    btnLink: {
        color: '#226597',
        marginTop: 20
    }
});


export default HomeScreen
