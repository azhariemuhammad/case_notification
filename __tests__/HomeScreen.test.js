import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import HomeScreen from "../App/screen/HomeScreen";



it("should render HomeScreen and compare with snapshot", () => {
    const wrapper = shallow(<HomeScreen/>)
    expect(wrapper).toMatchSnapshot()
})
