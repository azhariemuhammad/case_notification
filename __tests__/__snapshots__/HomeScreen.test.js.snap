// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HomeScreen and compare with snapshot 1`] = `ShallowWrapper {}`;
